Userland/Libraries/LibWeb/Streams/AbstractOperations.cpp

```diff
@@ -5,15 +5,33 @@
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
+#include <LibJS/Runtime/PromiseCapability.h>
+#include <LibJS/Runtime/PromiseConstructor.h>
 #include <LibWeb/Streams/AbstractOperations.h>
 #include <LibWeb/Streams/ReadableStream.h>
+#include <LibWeb/Streams/ReadableStreamDefaultController.h>
 #include <LibWeb/Streams/ReadableStreamDefaultReader.h>
 #include <LibWeb/Streams/ReadableStreamGenericReader.h>
 #include <LibWeb/WebIDL/ExceptionOr.h>
 #include <LibWeb/WebIDL/Promise.h>
 
 namespace Web::Streams {
 
+// https://webidl.spec.whatwg.org/#es-promise
+static WebIDL::ExceptionOr<JS::NonnullGCPtr<WebIDL::Promise>> convert_value_to_promise(JS::Realm& realm, JS::Value value)
+{
+    auto& vm = realm.vm();
+
+    // 1. Let promiseCapability be ? NewPromiseCapability(%Promise%).
+    auto promise_capability = TRY(JS::new_promise_capability(vm, realm.intrinsics().promise_constructor()));
```

Comment:
https://github.com/SerenityOS/serenity/blob/88d64fcb553fa2c95b0153649d5fcc1db99d75f8/Userland/Libraries/LibWeb/WebIDL/Promise.cpp#L27-L29

Reply:
Well now that you point it out, this `convert_value_to_promise` function is the exact same as `create_resolved_promise` :sweat_smile:

Reply:
This doesn't seem resolved. It sounds like you're saying we can use `create_resolved_promise` directly, and if not at the very least that TRY should be a MUST + NOTE